easybuild/easyblocks/o/openssl_wrapper.py

```diff
@@ -288,20 +282,23 @@ def __init__(self, *args, **kwargs):
             raise EasyBuildError(err_msg, self.version, self.system_ssl['version'])
 
         # Check system OpenSSL binary
-        if self.majmin_version == '1.1':
-            # prefer 'openssl11' over 'openssl' with v1.1
-            self.system_ssl['bin'] = which('openssl11')
-        elif LooseVersion(self.version) >= LooseVersion('3'):
-            # prefer 'openssl3' over 'openssl' with v3.x
-            self.system_ssl['bin'] = which('openssl3')
-
-        if not self.system_ssl['bin']:
-            self.system_ssl['bin'] = which('openssl')
+        target_ssl_bins = ['openssl']
+        if self.generation == '1.1':
+            target_ssl_bins.insert(0, 'openssl11')  # prefer 'openssl11' over 'openssl' with v1.1
+        elif self.generation == '3':
+            target_ssl_bins.insert(0, 'openssl3')  # prefer 'openssl3' over 'openssl' with v3
+
+        for target_bin in target_ssl_bins:
+            target_bin_path, target_bin_version = self.get_openssl_bin_version(target_bin)
+            if target_bin_version == self.system_ssl['version']:
```

Comment:
This check is too strict for Ubuntu.

Reply:
Ok, changed approach. Now we define a minimum **and** maximum version and only check that each component falls in that version range. Components may have different versions between them.

Reply:
Thanks a lot :+1: It is working now on Ubuntu 24.04 and 22.04.